maven-repository-metadata/src/main/java/org/apache/maven/artifact/repository/metadata/io/xpp3/MetadataXpp3Reader.java

```diff
@@ -26,22 +26,23 @@
 import java.io.Reader;
 
 import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 
 /**
- * Provide public methods from {@link org.apache.maven.artifact.repository.metadata.io.MetadataXpp3Reader}
+ * Provide public methods from {@link org.apache.maven.artifact.repository.metadata.io.MetadataStaxReader}
  *
  * @deprecated Maven 3 compatability
  */
 @Deprecated
```

Comment:
?

Reply:
You are right, we should point what is replacement in v4.

Reply:
@gnodet - Should we prefer `org.apache.maven.artifact.repository.metadata.io.MetadataStaxReader` for v4 or should be something added in api model.

Now everything is in `maven-repository-metadata` module

Reply:
@cstamas, @gnodet 
This depreciation was not introduced by this PR.

To remember and improvements I've create an issue: 
https://issues.apache.org/jira/browse/MNG-7903

Reply:
@gnodet is this deprecation actually a redirection to API? Or why are these classes deprecated? Also, how is Maven4 loading up metadata? Also, how is app using maven-repository-model as lib supposed to load metadata?

Reply:
The Xpp3 parsers/writers have been deprecated in favour of the Stax ones.  This is unrelated to any v3 / v4 API change.

Reply:
But this pr is about to change reader/writer from xpp to stax? So am confused now 🤔

Reply:
Ah, I think you're right.  Those two classes were removed when introducing the v4 api, later brought back to restore compatibility with Maven 3 but in a deprecated state.  So this is definitely a deprecation to switch to the v4 api.  Might be slightly premature though...
Anyway the layout is the same as for the maven model: the old v3 model for metadata is layered on top of the immutable one for v4.  Maybe the v4 generated model should be moved into the a `api/maven-api-metadata` module ?

Reply:
Maybe we should not add a technology used by parser in class name. 

When some days we figure out to use SuperFoo Xml parser we next change classes from MetadataStaxReader to MetadataSuperFooReader again?

Maybe name should be like MetadataXmlReader

Reply:
The v4 api offers an abstraction for xml parsing/writing when used inside Maven:
  https://github.com/apache/maven/tree/master/api/maven-api-core/src/main/java/org/apache/maven/api/services/xml

Reply:
So as a target and clean implementation should be:
-  org.apache.maven.api.services.xml.MetadataXmlFactory - in api/maven-api-core
- new module api/maven-api-metadata - with model

And then deprecated MetadataXpp3Reader should mention as replacement MetadataXmlFactory

I'm on right way?

Reply:
I think so.  Not completely sure about the MetadataXmlFactory as it is only available from inside maven. For outside maven (tooling reading the repositories somehow), we may want to point to the Stax reader too.  I guess both should be mentioned.